### DISCUSSION TITLE
How Does External Router Support Work?

### DISCUSSION THREAD

**Discussion by themajesticking** [tesla-android/issue-tracker#165] [Q&A] [answered]
Version 2022.45.1 now has support for external routers which I have been waiting for. I have my Pi connected to an LTE modem/router via ethernet which serves internet to both the Pi and the car. Before, this configuration provided internet access to the Pi and the car got internet access by tethering to the Pi's Wifi. This has worked fine and is still working but obviously connecting both the car and the Pi to the LTE modem would be more ideal. However, I can't figure out how the new external router support feature works. Here is the release note info:

Internet access
Support for external routers
Tesla Android webserver and other services can now be accessed externally using ethernet. This can be used to access the device in your home network or in the car with an external router.

I assume this means I should be able to access the web interface by typing the IP address of the Pi (the DHCP IP address from the router) into the Tesla web browser but that doesn't work. Is there a specific port we should be using or should I be going about this a different way?

**Comment by mikegapinski** [comment, marked as answer]
It works like you said, the pi can be accessed by the IP assigned by the router. If you want to access the pi from another router in the car keep in mind that the browser doesn't connect to standard local address ranges. Your router must assign an "external" ip to your pi, (like 3.3.3.x for example).

If your router has openwrt you can redirect traffic from an external address to your local one with iptables.

Keep in mind that you won't have offline mode anymore if you use an external router for connectivity, this is handled by the dns server on the pi. External router support was added after someone wanted to access the pi from their local network at home, in this case all address ranges work without issues

**Reply by themajesticking** [reply]
Thanks for the info. There is just one router in this configuration. Just a Netgear 5G modem/router. I grabbed the IP from the Android interface on the Pi and just used that. I don't think the Netgear 5G modem is sophisticated enough to configure iptables. I never used offline mode. I was getting internet from the modem over ethernet and the car was sharing modem's connection through the Pi. It sounds like you're saying my setup should physically stay the same but that the modem itself needs to be reconfigured somehow?

**Reply by mikegapinski** [reply]
Yes, change DHCP settings on the router to assign addresses from a public ip space. So instead of for instance: 192.168.x.x use 3.3.x.x

**Reply by themajesticking** [reply]
Okay I'm still a bit confused. I apologize for the confusion, networking isn't my strongest suit. I need to assign a public IP even though my car and the Pi are on the same LAN? Both of them are on the same router.  I'm not understanding why accessing the Pi at 192.168.x.x doesn't work.

**Reply by mikegapinski** [reply]
Tesla uses local IP addresses inside its own network, via automotive ethernet. That's why local ranges are blocked in the browser

**Reply by themajesticking** [reply]
Ahhh. Now it makes sense. I'll try that. Thanks again!

**Reply by GtHbbrr** [reply]
Did you get it to work in your car using the external router?

**Comment by themajesticking** [comment]
Changing DHCP range to something outside of 192.169.x.x worked perfectly!! Thanks!!
